axiom/datasets_integration_test.go

```diff
@@ -327,37 +327,57 @@ func (s *DatasetsTestSuite) TestCursor() {
 	// retrieved from the query status MinCursor or MaxCursor fields, depending
 	// on the queries sort order.
 	midRowID := queryResult.Matches[1].RowID
-	midTime := queryResult.Matches[1].Time
 
-	// Query events with a cursor in descending order.
+	// Query events with a cursor in descending order...
 	apl = fmt.Sprintf("['%s'] | sort by _time desc", s.dataset.ID)
 	queryResult, err = s.client.Datasets.Query(s.ctx, apl,
-		// HINT(lukasmalkmus): Start time is inclusive when sorting newest
-		// first.
-		query.SetStartTime(midTime.Add(-time.Nanosecond)),
+		query.SetStartTime(startTime),
+		query.SetEndTime(endTime),
+		query.SetCursor(midRowID, false),
+	)
+	s.Require().NoError(err)
+
+	if s.Len(queryResult.Matches, 1) {
+		s.Equal("bar", queryResult.Matches[0].Data["foo"])
```

Comment:
This is all a bit hard to follow, but I think it is correct as you have it. Maybe add a comment ala `// "buz", and "baz" skipped by cursor, so only oldest item left`